src/os/bluestore/NVMEDevice.cc

```diff
@@ -798,7 +798,7 @@ int NVMEDevice::open(string p)
     derr << __func__ << " unable to read " << p << ": " << cpp_strerror(r) << dendl;
     return r;
   }
-  while (r > 0 && !isalpha(buf[r-1])) {
+  while (r > 0 && !isxdigit(buf[r-1])) {
```

Comment:
wouldn't be the better approach to scan buf from the beginning while isxdigit? This will protect from all the potential garbage that might be at the end of the read buffer.